ball.js

```diff
@@ -0,0 +1,50 @@
+class Ball {
+  constructor(ctx, x, y, radius, color, speed) {
+    this.ctx = ctx;
+    this.x = x;
+    this.y = y;
+    this.radius = radius;
+    this.color = color;
+    this.speed = speed;
+    this.xDirection = 0;
+    this.yDirection = 1;
+  }
+
+  getTopEdge() {
+    return this.y - this.radius;
+  }
+
+  getBottomEdge() {
+    return this.y + this.radius;
+  }
+
+  getLeftEdge() {
+    return this.x - this.radius;
+  }
+
+  getRightEdge() {
+    return this.x + this.radius;
+  }
+
+  move() {
+    this.x += (this.xDirection * this.speed) / 2;
+    this.y += this.yDirection * this.speed;
+  }
+
+  setXDirection(direction) {
```

Comment:
you shouldn't be manually setting directions here. instead, you only ever be inverting directions. if a ball hits the right side of a brick, you invert the x direction to go the other way. same goes for every direction possible and Y

Reply:
At the start of the game, the ball's x direction is 0 as the ball is falling straight down towards the paddle. The options of direction values are not limited to -1 and 1, which is why I set the direction instead.

Reply:
if the game starts with the ball on the paddle, how does the ball go up? logically speaking, the ball only goes up because of the collision between ball and paddle.

Reply:
your collision logic should be what sets the ball in motion